### DISCUSSION TITLE
Bookmark Rules

### DISCUSSION THREAD

**Discussion by madprops** [madprops/grasshopper#132] [General]
Introducing simple rules that can automatically decide on which folder to save a bookmark:

![bookmarkrules234](https://github.com/user-attachments/assets/ed946d4b-9d0f-4818-9882-d80663892b82)

For now it only checks if the url of the item starts with the url of the rule, and then it uses the defined folder.

This is triggered on normal `Bookmark` commands.

This is on `General` settings.

**Comment by madprops** [comment]
```js
  let title

  if (url) {
    let rules = App.get_setting(`bookmark_rules`)

    for (let rule of rules) {
      if (url.startsWith(rule.url)) {
        title = rule.folder
        break
      }
    }
  }

  if (!title) {
    title = App.get_setting(`bookmarks_folder`)
  }

  if (!title) {
    title = App.get_default_setting(`bookmarks_folder`)
  }
```

**Comment by madprops** [comment]
A limitation it has is that it uses the title of a folder to find it, not the id, so it can be ambiguous if you have multiple folders with the same name, which could be the case with subfolders. So ideally you'd give it a unique name to the folders.

**Comment by N3C2L** [comment]
Great idea! Thanks!
Saves time because the bookmark folder popup won't be needed anymore if a rule was set before.

**Comment by madprops** [comment]
Second iteration looks like this, more options:

![bookmarkrules](https://github.com/user-attachments/assets/fd30d221-3eb3-48ea-9e99-7c28105b24df)

**Comment by N3C2L** [comment]
Thanks!
You already enhanced the domain rules with a cmd "domain rules" when executed from a selected tab. Do you think that's also possible for the bookmark rules also? Maybe with a folder popup where the selected folder will be auto inserted in the bookmark rule popup when executed from the tab. Then the URL from the tab will be inserted and the folder from the popup, both into the bookmark rule creation popup could be inserted in a fast way.

**Comment by madprops** [comment]
Next version will have this. You can use the `Bookmark Rule` command.

**Reply by N3C2L** [reply]
Awesome! Much appreciated!
